SEED-platform/seed
# Inventory groups frontend edits

#### Any background context you want to provide?

#### What's this PR do?
- Adds db constraint to inventory groups requiring unique name
- Adds validation to group creation. The group and it's inventory must share an ALI.
- Adds ability to choose an ALI if an empty group is created
- Adds Group Map page
- Adds temp partials and controllers for inventory group dashboard, systems & services, and meters
- Various improvements

#### How should this be manually tested?
Sample test files
[ali_tree_better_buildings_lite.xlsx](https://github.com/user-attachments/files/17212873/ali_tree_better_buildings_lite.xlsx)
[AH_BB_250props100_lat_long.xlsx](https://github.com/user-attachments/files/17212843/AH_BB_250props100_lat_long.xlsx) 
[AH_BB_250props100_lat_long_diag.xlsx](https://github.com/user-attachments/files/17212842/AH_BB_250props100_lat_long_diag.xlsx) - Same properties, but properties lat has been updated

- Upload property info with ALI and lat/long info to cycle 1. Upload 2nd file to cycle 2.
- Go to  Inventory List > Groups > Create Group > create 2 groups for 2 different leaf nodes node
- Go to Inventory List > Select properties of a leaf node > Actions > Add to/Remove from Groups > Add button for the group with a matching ALI should be enabled, and the group with a mismatched ALI should be disabled
- Select properties from different ALIs > Actions > Add to/Remove from Groups > should see a warning that the selected views do not meet group requirements. Create is disabled, all adds are disabled.
- Go to Groups > click group name > navigate through Inventory Group Detail pages. The "Properties" and "Map" tab cycle dropdowns should update data to reflect cycle inventory

#### What are the relevant tickets?

#### Screenshots (if appropriate)


![Screenshot 2024-10-01 at 9 54 05 AM](https://github.com/user-attachments/assets/6d8b27c6-d0b2-4679-afed-9b9ad930a3e9)
![Screenshot 2024-10-01 at 9 53 24 AM](https://github.com/user-attachments/assets/e9e2562e-a6c4-4d38-a787-012d13bb40c1)
![Screenshot 2024-10-01 at 9 55 57 AM](https://github.com/user-attachments/assets/d426051a-48bc-446e-b183-0c7cb66d22a2)
![Screenshot 2024-10-01 at 9 54 22 AM](https://github.com/user-attachments/assets/90c86e8e-02a0-46a7-a7bc-d39e5551f088)
